drape_frontend/line_shape.cpp

```diff
@@ -352,14 +347,47 @@ void LineShape::Construct(TBuilder & builder) const
   ASSERT(false, ("No implementation"));
 }
 
+template <class FnT> void LineShape::ForEachSplineSection(FnT && fn) const
+{
+  std::vector<m2::PointD> const & path = m_spline->GetPath();
+  ASSERT(!path.empty(), ());
+  size_t const sz = path.size() - 1;
+
+  for (size_t i = 0, j = 1; j <= sz; ++j)
+  {
+    /// @todo Make this kind of fitration in Spline?
+    if (path[i].EqualDxDy(path[j], kMwmPointAccuracy) && j < sz)
```

Comment:
when is this possible? for user tracks only? looks like its almost duplicating the filtering in apply_feature_functors